packages/backend/src/graphql/schema.graphql

```diff
@@ -111,6 +111,24 @@ type Query {
     id: String!
   ): IncomingPayment
 
+  "Fetch a paginated list of incoming payments."
+  incomingPayments(
+    "Forward pagination: Cursor (incoming payment ID) to start retrieving incoming payments after this point."
+    after: String
+    "Backward pagination: Cursor (outgoing payment ID) to start retrieving incoming payments before this point."
+    before: String
+    "Forward pagination: Limit the result to the first **n** incoming payments after the `after` cursor."
+    first: Int
+    "Backward pagination: Limit the result to the last **n** incoming payments before the `before` cursor."
+    last: Int
+    "Specify the sort order of incoming payments based on their creation date, either ascending or descending."
+    sortOrder: SortOrder
+    "Filter incoming payments based on specific criteria such as initiation reason."
+    filter: IncomingPaymentFilter
```

Comment:
Let's add this filter to the `walletAddress.incomingPayments` resolver (since that is the resolver we will be using for the POS service API)